code/perl/search_filters2.pl

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use JSON::MaybeXS;
+use Search::Xapian ':all';
+use FindBin qw($Bin);
+use lib $Bin;
+use Support;
+use Data::Dumper;
+
+
+my ($db_path, $query_string) = @ARGV;
+die "Usage: $0 DB_PATH QUERYTERM..." unless $db_path && $query_string;
```

Comment:
This ignores any arguments after the second - the code in `search_sorting.py` looks better.